analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -78,12 +79,14 @@ protected sealed override SymbolReferenceInfo CreateMessage(SyntaxTree syntaxTre
             return symbolReferenceInfo;
         }
 
+        protected override bool ShouldGenerateMetrics(SyntaxTree tree) =>
```

Comment:
This can be reverted and made `private` in base class after addressing the comment about `HasTooManyTokens`